### PR TITLE
Fixed inconsistencies in msg signer and pubtools-sign

### PR DESCRIPTION
**Pull request by midnightercz**

### CODE REVIEW

File: src/pubtools/sign/signers/msgsigner.py [hunk 1]
```diff
@@ -225,7 +230,7 @@ def clear_sign(self: MsgSigner, operation: ClearSignOperation):
         message_to_data = {}
         for in_data in operation.inputs:
             message = self._create_msg_message(
-                in_data,
+                base64.b64encode(in_data.encode("latin1")),
```
Comment (emilyzheng) [inline src/pubtools/sign/signers/msgsigner.py:233]: json.dumps fails on this byte string.

File: src/pubtools/sign/signers/msgsigner.py [hunk 2]
```diff
@@ -132,12 +132,17 @@ class MsgSigner(Signer):
...
 
     def _construct_signing_message(
-        self: MsgSigner, claim, signing_key, extra_attrs: Optional[Dict] = None
+        self: MsgSigner,
+        claim,
+        signing_key,
+        extra_attrs: Optional[Dict] = None,
+        sig_type: str = "container_signature",
     ):
+        data_attr = "claim_file" if sig_type == "container_signature" else "data"
```
Comment (querti) [inline src/pubtools/sign/signers/msgsigner.py:141]: This seems like an odd way to implement it, Maybe sig_type should be a boolean?

**Review by querti — CHANGES_REQUESTED** [review]

**Review by emilyzheng — APPROVED** [review]

File: src/pubtools/sign/signers/msgsigner.py (hunk 2 above)
Comment (midnightercz) [reply src/pubtools/sign/signers/msgsigner.py:141]: Why do you find it strange? sig_type is directly used as mtype in messaging headers. Also bool wouldn't allow to implement potential next signature type in the future

Comment (querti) [reply src/pubtools/sign/signers/msgsigner.py:141]: If multiple types will be allowed, shouldn't it be an enum? You can't just set it to any string and expect it to work, right? The possible values are not documented anywhere either.

**Review by querti — CHANGES_REQUESTED** [review]

**Review by querti — APPROVED** [review]

**Pull request merged**